server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/GetMessagesMethodStepdefs.java

```diff
@@ -246,6 +250,42 @@ public void appendMessageWithSpecialCase(String messageName, String mailbox) thr
         appendMessage(messageName, "eml/htmlWithLongAndComplicatedContent.eml");
     }
 
+    @Given("^the user has a message \"([^\"]*)\" in the \"([^\"]*)\" mailbox with flags \"([^\"]*)\"$")
+    public void appendMessageWithFlags(String messageName, String mailbox, List<String> keywords) throws Exception {
+        Flags flags = getFlagsFromListOfKeywords(keywords);
+        appendMessage(messageName, flags);
+    }
+
+    private Flags getFlagsFromListOfKeywords(List<String> keywords) {
+        return Keywords.factory()
+            .fromSet(keywords.stream()
+                .map(Keyword::new)
+                .collect(Guavate.toImmutableSet()))
+            .asFlags();
+    }
+
+    @Given("^the user has a message \"([^\"]*)\" in the \"([^\"]*)\" mailbox with Recent and Deleted flags \"([^\"]*)\"$")
+    public void appendMessageWithUnsupportedJmapFlags(String messageName, String mailbox, List<String> keywords) throws Exception {
```

Comment:
I don't see what is different from *appendMessageWithFlags*. Did you duplicate the methods?